sqsutil/datafetchers/numia_apr_fetcher.go

```diff
@@ -29,22 +30,24 @@ func GetFetchPoolAPRsFromNumiaCb(numiaHTTPClient passthroughdomain.NumiaHTTPClie
 			poolAPRsMap[poolAPR.PoolID] = poolAPR
 		}
 
-		return poolAPRsMap
+		return poolAPRsMap, nil
 	}
 }
 
 // GetFetchPoolPoolFeesFromTimeseries returns a callback to fetch pool fees from timeseries data stack.
 // It increments the error counter if the pool fees fetching fails.
 // It returns a callback function that returns the pool fees on success.
-func GetFetchPoolPoolFeesFromTimeseries(timeseriesHTTPClient passthroughdomain.TimeSeriesHTTPClient, logger log.Logger) func() map[uint64]passthroughdomain.PoolFee {
-	return func() map[uint64]passthroughdomain.PoolFee {
+func GetFetchPoolPoolFeesFromTimeseries(timeseriesHTTPClient passthroughdomain.TimeSeriesHTTPClient, logger log.Logger) func() (map[uint64]passthroughdomain.PoolFee, error) {
+	return func() (map[uint64]passthroughdomain.PoolFee, error) {
 		// Fetch pool APRs from the passthrough grpc client
 		poolFees, err := timeseriesHTTPClient.GetPoolFees()
 		if err != nil {
 			logger.Error("Failed to fetch pool fees", zap.Error(err))
 
 			// Increment the error counter
 			domain.SQSPassthroughTimeseriesPoolFeesFetchErrorCounter.Inc()
+
+			return nil, err
```

Comment:
Note: one of the reasons for the panic was the lack of return here.

The system would attempt to dereference a nil pointer `poolFees.Data` and crash.